ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/BonsaiLayeredWorldState.java

```diff
@@ -242,7 +243,7 @@ public Account get(final Address address) {
   }
 
   @Override
-  public Optional<Code> getContract(final Account account) {
+  public <T extends AccountState> Optional<Code> getContract(final T account) {
```

Comment:
Why the generic?  Account already implements AccountState.  You can do the getCodeHash at line 44 in CodeCahce without changing any types.

Reply:
brainbroke, up too late last night. fixed.